remix-app/app/routes/tasks.edit.$taskId/route.tsx

```diff
@@ -0,0 +1,63 @@
+import { json, LoaderFunctionArgs, ActionFunctionArgs } from "@remix-run/node";
+import { Form, useLoaderData, redirect } from "@remix-run/react";
+
+import { PrismaClient, Status } from '@prisma/client';
+
+const statusOptions = Object.values(Status);
+const prisma = new PrismaClient();
+
+export default function Page() {
+    const { task } = useLoaderData<typeof loader>();
+
+    return (
+        <div>
+            <h1 className="text-3xl font-bold">Edit Task</h1>
+            <Form method="put">
+                <label>
+                    <input type="text" name="title" className="block border-2" placeholder="Title" defaultValue={task?.title}/>
```

Comment:
```suggestion
                    <input type="text" name="title" className="block border-2" placeholder="Title" defaultValue={task.title}/>
```

ここもzodのバリデーションを挟めば解決するとは思いますが
オプショナルな部分をどうするかを考えると良いかもです。
例えば、必須の項目が足りていない場合はデータ不良として画面に出すとかとか。